external-gateways/stellar/src/lib.rs

```diff
@@ -326,12 +327,26 @@ impl TryFrom<(VerifierSet, Vec<SignerWithSig>)> for Proof {
     type Error = Report<Error>;
 
     fn try_from(
-        (verifier_set, mut signers): (VerifierSet, Vec<SignerWithSig>),
+        (verifier_set, signers): (VerifierSet, Vec<SignerWithSig>),
     ) -> Result<Self, Self::Error> {
-        signers.sort_by(|signer1, signer2| signer1.signer.pub_key.cmp(&signer2.signer.pub_key));
+        let mut signatures_by_signers: HashMap<_, _> = signers
```

Comment:
```suggestion
        let mut signatures_by_pub_keys: HashMap<_, _> = signers
```
signature by public key?